modules/ROOT/pages/database/database-connector-examples.adoc

```diff
@@ -0,0 +1,737 @@
+    </db:sql>
+    <db:input-parameters>#[{
+        name: 'Coffee Machine',
+        description: 'Model:XYZ99. Uses small size capsules.',
+        price: 120,
+        discount: 5}]
+    </db:input-parameters>
+</db:insert>
+----
+=============
+
+The insert operation outputs a `Statement Result` object that contains two fields: an _affectedRows_
+integer that indicates how many rows were affected by the query, and a _generatedKeys_ map
+that contains the autogenerated keys. Given this example, the output is as follows:
+
+[source,json,linenums]
+----
+{
+  "affectedRows": 1,
+  "generatedKeys": {
+
+  }
+}
+----
+
+The _generatedKeys_ map is empty, although a row is successfully added to the table.
+The map is empty because this information is disabled by default to avoid overhead.
+To activate the information and generate keys, add the `autoGenerateKeys` parameter, and then set it to true:
+
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig" autoGenerateKeys="true">
+   <db:sql>
+       INSERT INTO electronic(name, description, price, discount)
+       VALUES (:name, :description, :price, :discount)
+   </db:sql>
+   <db:input-parameters>#[{
+       name: 'Coffee Machine',
+       description: 'Model:XYZ99. Uses small size capsules.',
+       price: 120,
+       discount: 5}]
+   </db:input-parameters>
+</db:insert>
+----
+
+If the statement generates multiple keys per affected row, you can specify which columns
+should be returned by setting the parameter `autoGeneratedKeysColumnNames` to a list that contains
+the desired column names. For example, to ensure that only the _id_ column is
+returned, set the parameter to a DataWeave list with a single string element:
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig" autoGenerateKeys="true" autoGeneratedKeysColumnNames="#[['id']]">
+    ...
+</db:insert>
+----
+
+== Update Example
+
+Given the table that is defined in the insert example, the operation to update the *electronic* table
+to set the discount to 10 percent for all items that have a price value above 100 is as follows:
+
+[literal,source,xml,linenums]
+....
+<db:update config-ref="dbConfig">
+    <db:sql><![CDATA[#["UPDATE electronic SET discount = :discount WHERE price > :price"]]]></db:sql>
+	<db:input-parameters>#[{
+        discount: 10,
+        price: 100
+    }]</db:input-parameters>
+</db:update>
+....
+
+The `<![CDATA[...]]>` wrapper allows you to use special characters, such as `>` or `"`, in the query.
+Otherwise, you must use XML-escaped versions of those characters, such as `+&gt;+` and  `+&quot;+`.
+
+When you use the Anypoint Studio visual user interface for connectors instead of the XML code view, you can type special
+characters directly into the `SQL Query Text` box, and Anypoint Studio automatically
+changes the character to its escaped version in the XML view.
+
+== Delete Example
+
+Given the table defined in the <<_insert_example>>, the following operation
```

Comment:
formatting?

Reply:
Removed formatting. Was making a reference to the table made in the Insert section. Named that since I am not doing an in-topic reference.